gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/converters/QueryAndGatewayDtoConverter.java

```diff
@@ -570,4 +570,41 @@ public static org.hypertrace.gateway.service.v1.common.Value convertToGatewayVal
     return QueryAndGatewayDtoConverter.convertQueryValueToGatewayValue(
         queryValue, attributeMetadata);
   }
+
+  public static org.hypertrace.gateway.service.v1.common.Value
+      convertToGatewayValueForMapAttributeValue(
+          AttributeKind attributeKind,
+          Map<String, AttributeMetadata> resultKeyToAttributeMetadataMap,
+          ColumnMetadata metadata,
+          Value queryValue) {
+    String resultKey = metadata.getColumnName();
+    // if there's no override from the map, then this is regular attribute, then use the attribute
+    // map to convert the value
+    AttributeMetadata attributeMetadata;
+    if (attributeKind == null) {
+      attributeMetadata = resultKeyToAttributeMetadataMap.get(resultKey);
+    } else {
+      switch (attributeKind) {
+        case TYPE_STRING_MAP:
```

Comment:
So after all this, we assume it would be a string - which makes sense - but do we really need all that complexity given that assumption? It seems like we really only need to accept the attribute kind and default to string for when we can't look it up.

Reply:
I tried to design it a bit generically to support additional maps in future.

Which complexity though? Passing around the attribute expression, or mapping the alias to attribute map expression?

Reply:
both - I would think that we could just pass in the type, and when we lookup, if we can't map an attribute expression to  metadata (or more specifically, it has a subpath and the key maps to a map type) we just can default to string. That doesn't seem like it requires holding state in the request context or anything.

For other types of maps, we could add similar logic - e.g number map + subpath => number

Reply:
I didn't quite follow. How would you know it's an attribute expression, without passing in the attribute expression around?

And, in order to get handle of the attribute expression, you would need to map the alias to attribute expression in request context, while building the request. And, then get the attribute expression from request context, while building the response

Reply:
discussed this with @skjindal93 offline and concluded on an alternative approach to attempt to avoid holding state in context and generally reduce complexity - update the conversion methods to work with an AttributeKind (which is all they read off of metadata, anyway), rather than an AttributeMetadata. This will allow all types, map or otherwise, to work consistently and be resolved at the same time/place.